tonic/src/transport/server/incoming.rs

```diff
@@ -27,7 +27,15 @@ where
         let mut incoming = pin!(incoming);
 
         while let Some(item) = incoming.next().await {
-            yield item.map(ServerIo::new_io)?
+            if item.is_err() {
```

Comment:
Can't we do `if let Err(e) = item {` here?

IMO we should try a `downcast_ref()` to `io::Error` and then look for the right `io::ErrorKind`, would that be feasible?